lib/src/repo.rs

```diff
@@ -967,3 +967,24 @@ impl MutableRepo {
 #[derive(Debug, Copy, Clone, Error)]
 #[error("Cannot rewrite the root commit")]
 pub struct RewriteRootCommit;
+
+#[derive(Debug, Error)]
+#[error("Cannot access {path}")]
+pub struct PathError {
+    pub path: PathBuf,
+    #[source]
+    pub error: io::Error,
+}
+
+pub(crate) trait IoResultExt<T> {
+    fn path<P: AsRef<Path>>(self, path: P) -> Result<T, PathError>;
```

Comment:
`path()` is not very clear at the call-sites. Something like `error_context()` seems clearer. Feel free to pick something else.

Reply:
Went with `context` to match anyhow's precedent.